src/services/codefixes/helpers.ts

```diff
@@ -218,16 +218,26 @@ namespace ts.codefix {
         methodName: string,
         inJs: boolean,
         makeStatic: boolean,
-        preferences: UserPreferences,
         contextNode: Node,
+        importAdder: ImportAdder
     ): MethodDeclaration {
         const body = !isInterfaceDeclaration(contextNode);
         const { typeArguments, arguments: args, parent } = call;
+        const scriptTarget = getEmitScriptTarget(context.program.getCompilerOptions());
         const checker = context.program.getTypeChecker();
         const tracker = getNoopSymbolTrackerWithResolver(context);
-        const types = map(args, arg =>
-            // Widen the type so we don't emit nonsense annotations like "function fn(x: 3) {"
-            checker.typeToTypeNode(checker.getBaseTypeOfLiteralType(checker.getTypeAtLocation(arg)), contextNode, /*flags*/ undefined, tracker));
+        const types = map(args, arg => {
+            const type = checker.getBaseTypeOfLiteralType(checker.getTypeAtLocation(arg));
+            const typeNode = checker.typeToTypeNode(type, contextNode, /*flags*/ undefined, tracker);
+            if (typeNode?.kind === SyntaxKind.ImportType) {

```

Comment:
Say you get back `import("a").Cls<import("b").Other>` - shouldn't you visit the type node tree and attempt to autoimport all the import type nodes, rather than just the top-level one?

Reply:
@weswigham Thanks for the feedback 👍. I've added additional tests.